pytorch_lightning/strategies/collaborative.py

```diff
@@ -0,0 +1,504 @@
+import http
+import ipaddress
+import logging
+import os
+import re
+import threading
+import time
+import warnings
+from http.server import BaseHTTPRequestHandler
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import requests
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies.strategy import Strategy, TBroadcast
+from pytorch_lightning.utilities import rank_zero_only, rank_zero_warn
+from pytorch_lightning.utilities.data import extract_batch_size
+from pytorch_lightning.utilities.enums import PrecisionType
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _HIVEMIND_AVAILABLE
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.types import LRSchedulerTypeUnion
+
+if _HIVEMIND_AVAILABLE:
+    import hivemind
+
+log = logging.getLogger(__name__)
+
+
+class CollaborativeStrategy(Strategy):
+    def __init__(
+        self,
+        target_batch_size: int,
+        run_id: str = "lightning_run",
+        batch_size: Optional[int] = None,
+        delay_state_averaging: bool = False,
+        delay_optimizer_step: Optional[bool] = None,
+        delay_grad_averaging: bool = False,
+        offload_optimizer: Optional[bool] = None,
+        reuse_grad_buffers: bool = False,
+        scheduler_fn: Optional[Callable] = None,
+        matchmaking_time: float = 5.0,
+        averaging_timeout: float = 30.0,
+        verbose: bool = True,
+        averager_opts: Optional[Dict] = None,
+        host_maddrs: Optional[List] = None,
+        initial_peers: Optional[Union[str, List]] = None,
+        endpoint: Optional[bool] = None,
+        peer_endpoint: Optional[str] = None,
+        persistent: bool = True,
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        retry_initial_peers: int = 5,
+        retry_peer_sleep_duration: int = 5,
+        **optimizer_kwargs: Any,
+    ):
+        """Provides capabilities to train using the Hivemind Library, training collaboratively across the internet
+        with unreliable machines. `For more information: https://pytorch-
+        lightning.readthedocs.io/en/latest/strategies/collaborative_training.html`.
+
+        .. warning:: ``CollaborativeStrategy`` is experimental and subject to change.
+
+        Arguments:
```

Comment:
I think this needs to be `Args`:
```suggestion
        Args:
```

Reply:
Looking at the other strategies they all follow the `Arguments` name, am I missing something?

Reply:
Ah, looks like both `Args` and `Arguments` are ok, although the google style uses `Args`. Within PL usage is just inconsistent haha

Reply:
So I think this can be ignored :smiley: